src/pages/commits/HistoryTableIcon/index.tsx

```diff
@@ -0,0 +1,60 @@
+import styled from "@emotion/styled";
+import Tooltip from "@leafygreen-ui/tooltip";
+import { Body } from "@leafygreen-ui/typography";
+import { TaskStatusIcon } from "components/TaskStatusIcon";
+import { TaskStatus } from "types/task";
+
+interface HistoryTableIconProps {
+  status: TaskStatus;
+  label?: string;
+  failingTests?: string[];
+  inactive?: boolean;
+}
+
+export const HistoryTableIcon: React.FC<HistoryTableIconProps> = ({
+  status,
+  label,
+  failingTests = [],
+  inactive,
+}) => (
+  <Tooltip
+    usePortal={false}
+    align="right"
+    justify="middle"
+    enabled={!inactive && !!failingTests.length}
+    popoverZIndex={1}
+    trigger={
+      <Container>
+        <IconContainer inactive={inactive}>
+          <TaskStatusIcon status={status} />
+        </IconContainer>
+        {!inactive && <Body>{label}</Body>}
+      </Container>
+    }
+    triggerEvent="hover"
+  >
+    <TestNameContainer>
+      {failingTests.map((testName) => (
+        <Body>{testName}</Body>
+      ))}
+    </TestNameContainer>
+  </Tooltip>
+);
+
+const TestNameContainer = styled.div`
+  white-space: nowrap;
+`;
+const Container = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+`;
+
+interface IconContainerProps {
+  inactive?: boolean;
+}
+const IconContainer = styled.div<IconContainerProps>`
+  ${({ inactive }) => inactive && "opacity: .4;"}
+  height: 30px;
+  width: 30px;
```

Comment:
Currently some of the icons are different heights ranging between 28-30px. We should be able to remove this height and width after I upload updated svgs.

Reply:
Would it be possible for us to choose a specific width when using the TaskStatusIcons? Asking because there is [this comment](https://www.figma.com/file/TxlYD1vlJPgZ8S2Z8sZfe9/Mainline-Commit-UI?node-id=0%3A1) on the Figma.

Reply:
We can choose the size of the icon after switching over to using the leafygreen Icon component instead of rendering the SVG directly
https://github.com/evergreen-ci/spruce/pull/839/files#diff-99f280eb2b949701a956ff9ebb2408253a12a15d650f42b0ba7bba675ba80b02R16